src/main/java/compsci4039/week2/people/Person.java

```diff
@@ -0,0 +1,58 @@
+package compsci4039.week2.people;
+
+/*
+ * Programming IT, Laboratory Exercises
+ * 1 People
+ * Parts 1 - 3
+ * https://docs.google.com/document/d/13s7-_-iO2AV9_5gIA4YoxljJzs8TKGVcJHWeYaULihY
+ */
+
+public class Person {
+
+    private String name;
+    private int age;
+    private Gender gender;
+
+    public Person(String name, int age, Gender gender) {
+        this.name = name;
+        this.age = age;
+        this.gender = gender;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public int getAge() {
+        return age;
+    }
+
+    public void setAge(int age) {
+        this.age = age;
+    }
+
+    public Gender getGender() {
+        return this.gender;
+    }
+
+    public void setGender(Gender gender) {
+        this.setGender(gender);
+    }
+
+    public void sayHello() {
+        System.out.printf("Hi, I'm  %s and I am %d years old.", name, age);
+    }
+
+    public void reportProfile() {
+        System.out.printf("Hello, very nice to meet you!\nFor those who want to know, my name is %s.\nI am %s years old and identify my gender as being %s", this.name, this.age, this.gender);
+    }
+
+    @Override
```

Comment:
Not sure why you're overriding the method here, isn't that only for subclasses? Wouldn't this conflict with your 'Student' @Override?

https://www.geeksforgeeks.org/overriding-in-java/

Unless you can tell me otherwise! (I had to look it up...)

Reply:
Hi @gsears

Every ```class``` is actually an **object itself** 😮 

They are a very **special kind**, tho. In Java, they all inherit from the ```Object``` class (either directly or indirectly). As such, there is a default implementation in ```Object``` that prints the class name + the hascode. I suggest you create a simples class and experiment yourself 😄 

Therefore, if we want ```toString``` method to generate our custom String, **we need to override it**. Nopes, it won't conflict with the ```Student``` override because it has its own customization. 

When a method is annotated with ```@Override```, it ignores its parent's implementation (unless you call ```super``` inside and make some special operation). A great example of this concept is the constructor in ```Student```, which takes advantage of its parents constructor.

### Am I gonna get an ```approve``` now? 😝 

Further reading from the same source you quoted: 
https://www.geeksforgeeks.org/overriding-tostring-method-in-java/

Reply:
Gotchya, makes sense that java has a base 'Object' class. Interesting to know there's a toString method on it. And yeah, no conflict makes sense.

Cheers for clarification. ;)